### PR TITLE
5562: Add api call tests

### PR DESCRIPTION
**Pull request by kat1906**

### CODE REVIEW

File: app/pages/common/dashboard/controller.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import { ManifestProvider } from '../../../manifestProvider';
+import { getDashboardManifest } from '../../../manifestProvider';
 import { getData, putData } from '../../../apiProvider';
 import { createDashboardPageContext } from './createDashboardPageContext';
 
 export const getMarketingPageDashboardContext = async ({
   solutionId, validationErrors, userContextType = 'supplier',
 }) => {
-  const dashboardManifest = new ManifestProvider().getDashboardManifest({ userContextType });
+  const dashboardManifest = await getDashboardManifest({ userContextType });
```
Comment (alibombaye) [inline app/pages/common/dashboard/controller.js:8]: You don't need the await here

File: app/pages/common/dashboard/controller.test.js [hunk 2]
```diff
@@ -1,75 +1,178 @@
...
+        await getMarketingPageDashboardContext({ solutionId: 'some-solution-id', dashboardId: 'some-dashboard-id' });
+
+        expect(context.createDashboardPageContext.mock.calls.length).toEqual(1);
+        expect(context.createDashboardPageContext).toHaveBeenCalledWith({
+          solutionId: 'some-solution-id',
+          solutionName: dashboardData.name,
+          supplierName: dashboardData['supplier-name'],
+          dashboardManifest,
+          marketingDataSections: dashboardData.sections,
+          validationErrors: undefined,
```
Comment (alibombaye) [inline app/pages/common/dashboard/controller.test.js:89]: can remove this from the check

Comment (kat1906) [reply app/pages/common/dashboard/controller.test.js:89]: no I can't as it checks the whole object and this is what is passed through if there are no validation errors

File: app/pages/common/dashboard/controller.test.js [hunk 3]
```diff
@@ -1,75 +1,178 @@
...
+        });
+
+        expect(context.createDashboardPageContext.mock.calls.length).toEqual(1);
+        expect(context.createDashboardPageContext).toHaveBeenCalledWith({
+          solutionId: 'some-solution-id',
+          solutionName: dashboardData.name,
+          supplierName: dashboardData['supplier-name'],
+          dashboardManifest,
+          marketingDataSections: dashboardData.sections,
+          validationErrors: undefined,
```
Comment (alibombaye) [inline app/pages/common/dashboard/controller.test.js:152]: remove if not needed

Comment (kat1906) [reply app/pages/common/dashboard/controller.test.js:152]: it is needed

File: app/pages/supplier/dashboard/subDashboards/controller.js [hunk 4]
```diff
@@ -1,9 +1,9 @@
-import { ManifestProvider } from '../../../../manifestProvider';
+import { getSubDashboardManifest } from '../../../../manifestProvider';
 import { getData } from '../../../../apiProvider';
 import { createDashboardPageContext } from '../../../common/dashboard/createDashboardPageContext';
 
 export const getSubDashboardPageContext = async ({ solutionId, dashboardId }) => {
-  const dashboardManifest = new ManifestProvider().getSubDashboardManifest({ dashboardId });
+  const dashboardManifest = await getSubDashboardManifest({ dashboardId });
```
Comment (alibombaye) [inline app/pages/supplier/dashboard/subDashboards/controller.js]: remove the await

File: app/pages/common/dashboard/controller.test.js (hunk 2 above)
Comment (alibombaye) [reply app/pages/common/dashboard/controller.test.js:89]: ok

**Pull request merged**